File: LeetCode/Medium/0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: x[1])
        ret = 0
        next_idx, left, right = 1, 0, 1

        while right < len(intervals):
            if intervals[left][1] > intervals[right][0]:
                ret += 1
                right += 1
                next_idx = right
            else:
                left = next_idx
                right += 1
                next_idx = right
        
        return ret

File: LeetCode/Medium/0435-non-overlapping-intervals/0435-non-overlapping-intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](vector<int> &first, vector<int> &second) -> bool        {
            return first[1] < second[1];
        });
        int ret = 0;
        int left = 0, right = 1, nextIdx = 1;

        while (right < intervals.size()){
            if (intervals[left][1] > intervals[right][0]) ++ret;
            else left = nextIdx;
            
            ++right;
            nextIdx = right;
        }
        
        return ret;
    }
};